ProcessLib/HydroMechanics/CreateHydroMechanicsProcess.cpp

```diff
@@ -183,18 +174,49 @@ std::unique_ptr<Process> createHydroMechanicsProcess(
     }
 
     // Reference temperature
-    const auto& reference_temperature =
+    double const reference_temperature =
         //! \ogs_file_param{prj__processes__process__HYDRO_MECHANICS__reference_temperature}
-        config.getConfigParameter<double>(
-            "reference_temperature", std::numeric_limits<double>::quiet_NaN());
+        config.getConfigParameter<double>("reference_temperature", 1);
```

Comment:
The default 1 (and similar 0 in other parameters) should either be replaced by NaN (as before), or (because this are required parameters) the be omitted.
No-implicit-magic-values-policy ;)